ApiIndex/ServiceModel.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright 2020, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf.Reflection;
+using System.Linq;
+using static Google.Cloud.Tools.ApiIndex.V1.Method.Types;
+
+namespace Google.Cloud.Tools.ApiIndex
+{
+    /// <summary>
+    /// Internal representation of a service, without reference to any specific schema version.
+    /// </summary>
+    public class ServiceModel
+    {
+        private ServiceDescriptor service;
+
+        public ServiceModel(ServiceDescriptor service)
+        {
+            this.service = service;
+        }
+
+        public V1.Service ToV1Service() => new V1.Service
+        {
+            FullName = service.FullName,
+            ShortName = service.Name,
+            Methods = { service.Methods.OrderBy(method => method.Name).Select(ConvertMethod) }
+        };
+
+        private static V1.Method ConvertMethod(MethodDescriptor method) => new V1.Method
```

Comment:
And if things are staying as they are, even though this method is private, it'd be good if it was called `ConvertToV1Method` for consistency.

Reply:
Or `ToProtoV1Method` in this case, perhaps. But yes, completely agree. (Or I could make it a private method within the `ToV1Proto` method, in which case it doesn't need the version part.)